src/MongoDB.Driver/FindOptions.cs

```diff
@@ -100,6 +125,15 @@ public CursorType CursorType
             set { _maxAwaitTime = value; }
         }
 
+        /// <summary>
+        /// Gets or sets the max scan.
+        /// </summary>
+        public int? MaxScan
```

Comment:
`MaxScan` is `Obsolete` in `FindOperation`. Not sure if I should deprecate it here as well

Reply:
Maybe find out why `MaxScan` is obsolete?

It would be unusual for a *new* property to be obsolete from the moment it was added...

Reply:
It was deprecated in server version 4.0.
Here is the ticket: https://jira.mongodb.org/browse/CSHARP-2213, respective drivers (https://jira.mongodb.org/browse/DRIVERS-443) and spec (https://jira.mongodb.org/browse/SPEC-1031) tickets.
I was a bit confused because I didn't see it being deprecated in other drivers when looking at Modifiers ticket. But I just found out that other drivers were doing these changes in different order: they first deprecated Modifiers and then deprecated `MaxScan` and `Snapshot`. We are doing it in reversed order. That explains why we need to deprecate a newly added field :)
I will update the code shortly.

Reply:
Rather than add a field that is already deprecated would it not be better to just not add it?